main.cpp

```diff
@@ -135,29 +135,41 @@ void ProcessMidiEvent(u32 dwParam1, bool mouse) {
         int miditype=(msg.msg&0xf0)>>4;
         unsigned char chan=msg.msg&0xf;			
 		lastchan=chan;
+
+		// ignore advanced midi messages
+		if (miditype < 0x8 || miditype > 0xe) return;
+
         if (miditype==8 || miditype==9) {
+			// handle note-on and note-off events
+
+			// update the highest detected note
             if (!mouse && msg.data1>highestnoteseen) highestnoteseen=msg.data1;
-            int shifted_note=msg.data1+127-topnote;
+
+            // handle top note bound
+			int shifted_note=msg.data1+127-topnote;
             if (msg.data1==topnote) msg.data1=127;
             if (msg.data1>=topnote-4 && msg.data1<topnote) {
-				// accept both note off and zero-velocity note events as key release
-                if (miditype==8 || (miditype==9 && msg.data2 == 0)) {
-                    // note up. release the shifted note if it is down.						
+				if (miditype==8 || (miditype==9 && msg.data2 == 0)) {
+                    // note up. release the shifted note if it is down.
```

Comment:
I wonder if it's worth forcing miditype to 8 if we detect miditype==9 && msg.data2==0, in other words, 'rewrite' such messages as legit noteups. I think the game (and other downstream OSC consumers) might appreciate midi2osc doing such a rewrite.

Reply:
oh hang on I see that exactly what you did above. I must have been reading a different version/diff. yes I like this much better.